### PR TITLE
ajd chapter 8

### PR DESCRIPTION
**Pull request by alexdeas**
ajd chapter 8

### CODE REVIEW

File: chapter08/ajd/exercise1.pl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+GetOptions(
+    'file=s' => \$file,
+    'scalar' => \$scalar,
+);
+
+if ( ( !defined $file || $file eq '' ) && !defined $scalar ) {
+  die( "Usage: $0 [--file|-f <output file>] [--scalar|-s]\n" );
+}
+
+if ( defined $file && -e $file ) {
```
Comment (Spinodal) [inline chapter08/ajd/exercise1.pl:20]: haven't you already determined that the file is defined at this point?

File: chapter08/ajd/exercise2.pl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  if ( $line =~ /^(.*):/ ) {
+
+    # Prep the file name
+    my $file = lc($1);
+    $file =~ s/\W//g;
+
+    # Check whether there's a handle for it
+    if (!defined $handles->{$file}) {
+      open ( $handles->{$file}, '>>', "$file.info" ) || die ( "Can't open '$file' for writing: $!" ) ;
```
Comment (ardavey) [inline chapter08/ajd/exercise2.pl:25]: I think we were supposed to overwrite the file if it already exists

File: chapter08/ajd/exercise2.pl [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+if ( !@ARGV ) {
+  die ( "Usage: $0 <island data file>\n" );
+}
+
+open ( my $input, '<', $ARGV[0] ) || die ( "Couldn't open input file\n" );
+
+while ( my $line = <$input> ) {
+
+  state $handles;
```
Comment (ardavey) [inline chapter08/ajd/exercise2.pl]: We're not getting any benefit from `state` here, are we?

File: chapter08/ajd/exercise3.pl [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    print_directory_contents( $handle );
+    closedir ( $handle );
+  }
+}
+
+sub print_directory_contents {
+
+  my ( $handle ) = @_;
+
+  foreach my $file ( grep !/^\./, readdir ( $handle ) ) {
```
Comment (ardavey) [inline chapter08/ajd/exercise3.pl:26]: Hidden file hater

**Pull request merged**